etc/cmake/options.cmake

```diff
@@ -32,228 +32,223 @@ option(OT_APP_RCP "enable RCP app" ON)
 if(OT_IP6_FRAGM)
-    list(APPEND OT_PRIVATE_DEFINES "OPENTHREAD_CONFIG_IP6_FRAGMENTATION_ENABLE=1")
+    target_compile_definitions(ot-config INTERFACE "OPENTHREAD_CONFIG_IP6_FRAGMENTATION_ENABLE=1")
 endif()
 
 option(OT_JAM_DETECTION "enable jam detection support")
 if(OT_JAM_DETECTION)
-    list(APPEND OT_PRIVATE_DEFINES "OPENTHREAD_CONFIG_JAM_DETECTION_ENABLE=1")
+    target_compile_definitions(ot-config INTERFACE "OPENTHREAD_CONFIG_JAM_DETECTION_ENABLE=1")
 endif()
 
 option(OT_JOINER "enable joiner support")
 if(OT_JOINER)
-    list(APPEND OT_PRIVATE_DEFINES "OPENTHREAD_CONFIG_JOINER_ENABLE=1")
+    target_compile_definitions(ot-config INTERFACE "OPENTHREAD_CONFIG_JOINER_ENABLE=1")
 endif()
 
 option(OT_LEGACY "enable legacy network support")
 if(OT_LEGACY)
-    list(APPEND OT_PRIVATE_DEFINES "OPENTHREAD_CONFIG_LEGACY_ENABLE=1")
+    target_compile_definitions(ot-config INTERFACE "OPENTHREAD_CONFIG_LEGACY_ENABLE=1")
 endif()
 
 option(OT_LINK_RAW "enable link raw service")
 if(OT_LINK_RAW)
-    list(APPEND OT_PRIVATE_DEFINES "OPENTHREAD_CONFIG_LINK_RAW_ENABLE=1")
+    target_compile_definitions(ot-config INTERFACE "OPENTHREAD_CONFIG_LINK_RAW_ENABLE=1")
 endif()
 
 option(OT_LOG_LEVEL_DYNAMIC "enable dynamic log level control")
 if(OT_LOG_LEVEL_DYNAMIC)
-    list(APPEND OT_PRIVATE_DEFINES "OPENTHREAD_CONFIG_LOG_LEVEL_DYNAMIC_ENABLE=1")
+    target_compile_definitions(ot-config INTERFACE "OPENTHREAD_CONFIG_LOG_LEVEL_DYNAMIC_ENABLE=1")
 endif()
 
 option(OT_MAC_FILTER "enable mac filter support")
 if(OT_MAC_FILTER)
-    list(APPEND OT_PRIVATE_DEFINES "OPENTHREAD_CONFIG_MAC_FILTER_ENABLE=1")
+    target_compile_definitions(ot-config INTERFACE "OPENTHREAD_CONFIG_MAC_FILTER_ENABLE=1")
 endif()
 
 option(OT_MLE_LONG_ROUTES "enable MLE long routes extension (experimental, breaks Thread conformance)")
 if(OT_MLE_LONG_ROUTES)
-    list(APPEND OT_PRIVATE_DEFINES "OPENTHREAD_CONFIG_MLE_LONG_ROUTES_ENABLE=1")
+    target_compile_definitions(ot-config INTERFACE "OPENTHREAD_CONFIG_MLE_LONG_ROUTES_ENABLE=1")
 endif()
 
 option(OT_MTD_NETDIAG "enable TMF network diagnostics on MTDs")
 if(OT_MTD_NETDIAG)
-    list(APPEND OT_PRIVATE_DEFINES "OPENTHREAD_CONFIG_TMF_NETWORK_DIAG_MTD_ENABLE=1")
+    target_compile_definitions(ot-config INTERFACE "OPENTHREAD_CONFIG_TMF_NETWORK_DIAG_MTD_ENABLE=1")
 endif()
 
 option(OT_MULTIPLE_INSTANCE "enable multiple instances")
 if(OT_MULTIPLE_INSTANCE)
-    list(APPEND OT_PRIVATE_DEFINES "OPENTHREAD_CONFIG_MULTIPLE_INSTANCE_ENABLE=1")
+    target_compile_definitions(ot-config INTERFACE "OPENTHREAD_CONFIG_MULTIPLE_INSTANCE_ENABLE=1")
 endif()
 
 option(OT_PLATFORM_NETIF "enable platform netif support")
 if(OT_PLATFORM_NETIF)
-    list(APPEND OT_PRIVATE_DEFINES "OPENTHREAD_CONFIG_PLATFORM_NETIF_ENABLE=1")
-endif()
-
-option(OT_NCP_VENDOR_HOOK_SOURCE "Enable vendor hook for NCP")
-if(OT_NCP_VENDOR_HOOK_SOURCE)
-    list(APPEND OT_PRIVATE_DEFINES "OPENTHREAD_ENABLE_NCP_VENDOR_HOOK=1")
```

Comment:
Is it intentional to remove the `OPENTHREAD_ENABLE_NCP_VENDOR_HOOK` definition?

Reply:
yes